### PR TITLE
2669 improve location accuracy for street newsflashes

### PR DESCRIPTION
**Pull request by tkalir**

### CODE REVIEW

**Comment by atalyaalon** [comment]
@tkalir no urgency here, whenever you have time - can you please check and fix Pylint, Black and failing tests?

File: requirements.txt [hunk 1]
```diff
@@ -53,3 +53,7 @@ swifter==1.3.4
 telebot==0.0.5
 selenium==4.11.2
 apache-airflow-client==2.6.2
+openai==1.45.0
+langchain==0.2.16
+langchain_openai==0.1.25
+python-dotenv
```
Comment (atalyaalon) [inline requirements.txt:59]: Is `python-dotenv` necessary?
Also, `SQLAlchemy==1.4` modification necessary?
(I think we should perform packages upgrade, but it will be in a different pr with suitable tests)

File: anyway/parsers/location_extraction.py [hunk 2]
```diff
@@ -282,11 +307,12 @@ def reverse_geocode_extract(latitude, longitude):
     try:
         gmaps = googlemaps.Client(key=secrets.get("GOOGLE_MAPS_KEY"))
         geocode_result = gmaps.reverse_geocode((latitude, longitude))
-
+        print(geocode_result)
```
Comment (atalyaalon) [inline anyway/parsers/location_extraction.py:310]: Can you remove all prints in this code?

**Review by atalyaalon — CHANGES_REQUESTED** [review]
@tkalir All in all looks good., added some small comments
In addition, in the case that location is on a "boarder" of two municipalities and nearest accidents might be in different "yishuv_name", this case is not handled.
I think this should be taken into account in the current solution which might modify it a bit.

**Comment by atalyaalon** [comment]
@ziv17 can you review this one as well?
I want to make sure I didn't miss anything in edge cases

**Comment by atalyaalon** [comment]
@tkalir I added key to secrets.
Note that github-secrets are not accessible for PRs from forks, therefore tests in this PR will fail, but in merging dev to master they will not.

File: anyway/llm.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+import tiktoken
+
+from langchain.output_parsers import EnumOutputParser
+from langchain_core.prompts import PromptTemplate
+from langchain_openai import ChatOpenAI
+import langchain
+from enum import Enum
+from anyway import secrets
+
+api_key = secrets.get("OPENAI_API_KEY")
```
Comment (atalyaalon) [inline anyway/llm.py:12]: I suggest inserting the secret into a function, to avoid tests failures in tests from forks

File: anyway/llm.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+client = OpenAI(api_key=api_key)
+
+langchain.debug = True
+model = ChatOpenAI(api_key=api_key, temperature=0)
+
+
+def match_streets_with_langchain(street_names, location):
```
Comment (ziv17) [inline anyway/llm.py:19]: Hi, I prefer to have type hints. It makes the reading much easier.

Comment (ziv17) [inline anyway/llm.py:19]: Is this code used?

**Review by ziv17 — COMMENTED** [review]
Hi @tkalir , Very nice!
I learned a lot 😊

1. I assume this PR is only the beginning, and probably it is possible to extract more information, like the yishuv, resolution, etc.
2. Is it possible to get the level of confidence the AI tool is has in its answer? In certain cases it may be more accurate than the lat / lon that we have.
